pkg/login/social/generic_oauth.go

```diff
@@ -209,6 +211,33 @@ func (s *SocialGenericOAuth) extractFromToken(token *oauth2.Token) *UserInfoJson
 		return nil
 	}
 
+	headerBytes, err := base64.RawURLEncoding.DecodeString(matched[1])
+	if err != nil {
+		s.log.Error("Error base64 decoding header", "header", matched[1], "error", err)
+		return nil
+	}
+
+	var header map[string]string
+	err = json.Unmarshal(headerBytes, &header)
+	if compression, ok := header["zip"]; ok {
+		if compression == "DEF" {
+			fr, err := zlib.NewReader(bytes.NewReader(rawJSON))
+			if err != nil {
+				s.log.Error("Error creating zlib reader", "error", err)
+				return nil
+			}
+			defer fr.Close()
+			rawJSON, err = ioutil.ReadAll(fr)
+			if err != nil {
+				s.log.Error("Error decompressing payload", "error", err)
+				return nil
+			}
+		} else {
+			s.log.Error("Unknown compression algorithm", "algorithm", compression)
+			return nil
+		}
+	}
+
```

Comment:
In Go it's idiomatic to return early, and I think we should warn instead of error when an unrecognized compression algorithm is encountered, since it's not really an error on our part:
```suggestion
	if compression, ok := header["zip"]; ok {
		if compression != "DEF" {
			s.log.Warn("Unknown compression algorithm", "algorithm", compression)
			return nil
		}
		
		fr, err := zlib.NewReader(bytes.NewReader(rawJSON))
		if err != nil {
			s.log.Error("Error creating zlib reader", "error", err)
			return nil
		}
		defer fr.Close()
		rawJSON, err = ioutil.ReadAll(fr)
		if err != nil {
			s.log.Error("Error decompressing payload", "error", err)
			return nil
		}
	}

```